server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/UserProvisioning.scala

```diff
@@ -0,0 +1,64 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.util.UUID
+
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.metrics.api.MetricFactory
+import org.apache.james.metrics.api.TimeMetric.ExecutionResult.DEFAULT_100_MS_THRESHOLD
+import org.apache.james.user.api.{AlreadyExistInUsersRepositoryException, UsersRepository, UsersRepositoryException}
+import reactor.core.scala.publisher.SMono
+
+class UserProvisioning @Inject() (usersRepository: UsersRepository, metricFactory: MetricFactory) {
+
+  def provisionUser(session: MailboxSession): SMono[Unit] =
+    if (session != null && !usersRepository.isReadOnly) {
+      SMono.fromCallable(() => createAccountIfNeeded(session))
+        .`then`
+    } else {
+      SMono.empty
+    }
+
+  private def createAccountIfNeeded(session: MailboxSession): Unit = {
+    val timeMetric = metricFactory.timer("JMAP-RFC-8621-user-provisioning")
```

Comment:
Let's use the same metric name than for draft so that we don't need to update our board?

Reply:
Thought maybe wanted to be able to see the differences in terms of perf during the transition between draft and rfc-8621?

Reply:
No, it is just to differentiate the metric with jmap draft, to make comparisons if we need to. If the name is bothering you we can change it?